src/f32/funcs.rs

```diff
@@ -139,6 +168,69 @@ pub(crate) mod sse2 {
         _mm_or_ps(result, _mm_castsi128_ps(test))
     }
 
+    #[inline]
+    #[allow(non_snake_case)]
+    pub(crate) const fn _MM_SHUFFLE(z: u32, y: u32, x: u32, w: u32) -> i32 {
+        // core::arch::x86_64::_MM_SHUFFLE requires nightly
+        ((z << 6) | (y << 4) | (x << 2) | w) as i32
+    }
+
+    /// From DirectXMath: XMVectorModAngles
+    /// 
+    /// Returns a vector whose components are the corresponding components of Angles modulo 2PI.
+    pub(crate) unsafe fn m128_mod_angles(angles: __m128) -> __m128 {
+        let v = _mm_mul_ps(angles, g_XMReciprocalTwoPi.m128);
+        let v = m128_round(v);
+        XM_FMADD_PS!(g_XMTwoPi.m128, v, angles) // XMVectorNegativeMultiplySubtract
+    }
+
+    /// From DirectXMath: XMVectorSin
+    pub(crate) unsafe fn m128_sin(v: __m128) -> __m128 {
```

Comment:
Would be good to add a test for this similar to the `scalar_acos` test to get an idea of the error.